### PR TITLE
oop school library

### PR DESCRIPTION
**Pull request by Isabellanamutenda**
### **OOP school library**
In this project I:

- Created a class Corrector with a method that makes the first letter is capital 
- In-person add a method validate_name
- This is my last request hope I have made all the changes

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -32,5 +32,5 @@ def can_use_services?
   end
 end
 
-me = Person.new(14, 'donard')
+me = Person.new(14, 'Isabel')
```
Comment (rindrajosia) [inline person.rb]: - [x] [REQUIRED] I got an error when I run your APP. Please, make sure that an argument is passed when you use the **correct_name** method.

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Good job so far!
There are some issues to revise before continuing to the next project but hang in there and read the comments below with care. You can do it! :muscle:

##  To highlight :+1: 

- Good job by implementing the Corrector Class and Person Class

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Isabellanamutenda) [reply person.rb]: Fixed

File: person.rb [hunk 2]
```diff
@@ -32,5 +32,5 @@ def can_use_services?
...
 p me.validate_name
```
Comment (safafa) [inline person.rb:36]: when I run your code I get this error : 
image
I think the problem is in line 25 `@name = @corrector.correct_name` the method correct_name is expecting an argument as defined in the corrector class.

- [ ] Kindly fix the issue by passing the @name as argument.

File: corrector.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Corrector
+  def correct_name(name)
+    name = name.capitalize
+    name[0, 10] if name.length > 10
+  end
```
Comment (safafa) [inline corrector.rb]: - The function will return `nil` if you pass a name where `name.length < 10` and that's not the expected behavior.

- [ ] Kindly fix the issue.

**Review by safafa — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED ♻️ 

Hi @Isabellanamutenda , Good job working on some of the changes requested by the previous reviewer 👏🏼 👏🏼 

But,  there are still some which aren't addressed yet.

## Required changes :

- [ ] Check the inline comments.

## Optional changes :
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: person.rb (hunk 2 above)
Comment (henatan99) [reply person.rb:36]: This issue has been addressed well. Great job! :clap: :+1:

File: corrector.rb (hunk 3 above)
Comment (henatan99) [reply corrector.rb]: This issue has been addressed well. Nice job! :+1: :+1:

**Review by henatan99 — APPROVED** [review]
Hi @Isabellanamutenda , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**